api/capi/include/nntrainer.h

```diff
@@ -357,6 +357,34 @@ int ml_train_dataset_create_with_generator(ml_train_dataset_h *dataset,
                                            ml_train_datagen_cb train_cb,
                                            ml_train_datagen_cb valid_cb,
                                            ml_train_datagen_cb test_cb);
+/**
+ * @brief Set @a user_data to be used on the dataset if applicable.
+ * @details Use this function to set custom data that is to be called within @a
+ * ml_train_datagen_cb from @a dataset created with @a
+ * ml_train_dataset_create_with_generator(dataset, train_cb, valid_cb, test_cb).
+ * @a train_user_data is passed when @a train_cb is called.
+ * @a valid_user_data is passed when @a valid_cb is called.
+ * @a test_user_data is passed when @a test_cb is called.
+ * @remarks @a train_user_data, @a valid_user_data, @a test_user_data must be
+ * released by the user after use.
+ *
+ * @since_tizen 6.5
+ * @param[in] dataset The NNTrainer dataset handle.
+ * @param[in] train_user_data user data to be passed when @a train_cb is being
+ * called. Can be null.
+ * @param[in] valid_user_data user data to be passed when @a valid_cb is being
+ * called. Can be null.
+ * @param[in] test_user_data user data to be passed when @a test_cb is being
+ * called. Can be null.
+ * @return @c 0 on successs. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successsful
+ * @retval #ML_ERROR_INVALID_PARAMETER if @a dataset is not created from @a
+ * ml_train_dataset_create_with_generator
+ */
+int ml_train_dataset_generator_set_user_data(ml_train_dataset_h *dataset,
```

Comment:
If you are adding this, please revert the changes from https://github.com/nnstreamer/nntrainer/pull/533

Reply:
Agh.. I forgot we had that functionality already.

I think #533 style is better, in this case, we just have to change the document.

I wonder if we can split those into train_user_data, valid_user_data, ... as far as I remember there was a problem with this.
"ml_train_dataset_set_property(dataset, "train_user_data", (void *) data, " NULL)"

Reply:
Yes, we have to support 3 user data corresponding to each train, valid and test.
That will have to added w.r.t. #533.

Reply:
We discussed offline to add something like "train_user_data", "validation_user_data", "test_user_data" w.r.t #533